app/controllers/finance/ordergroups_controller.rb

```diff
@@ -1,20 +1,27 @@
 class Finance::OrdergroupsController < Finance::BaseController
 
   def index
-    if params["sort"]
-      sort = case params["sort"]
-               when "name" then "name"
-               when "account_balance" then "account_balance"
-               when "name_reverse" then "name DESC"
-               when "account_balance_reverse" then "account_balance DESC"
-             end
+    m = /^(?<col>name|sum_of_class_\d+)(?<reverse>_reverse)?$/.match params["sort"]
+    if m
+      sort = m[:col]
+      sort += ' DESC' if m[:reverse]
     else
       sort = "name"
     end
 
     @ordergroups = Ordergroup.undeleted.order(sort)
-    @ordergroups = @ordergroups.where('name LIKE ?', "%#{params[:query]}%") unless params[:query].nil?
+    @ordergroups = @ordergroups.where('groups.name LIKE ?', "%#{params[:query]}%") unless params[:query].nil?
 
     @ordergroups = @ordergroups.page(params[:page]).per(@per_page)
+
+    select = ['groups.*']
+    FinancialTransactionClass.all.each do |c|
+      select << "sum(CASE financial_transaction_types.financial_transaction_class_id WHEN #{c.id} THEN financial_transactions.amount ELSE 0 END) AS sum_of_class_#{c.id}"
+    end
```

Comment:
I wonder if this could become one SQL statement. Perhaps not if we want to keep compatible with MySQL and PostgreSQL. At the very least, we should move this to a helper method, or rather to the model.